File: LeetCode/cpp/896.cpp
class Solution {
public:
    bool isMonotonic(vector<int> &A) {
        auto size = A.size();
        int &&inc = 0;
        for (int i = 0; i < size - 1; ++i) {
            if (A[i] < A[i + 1]) {
                if (inc == -1)
                    return false;
                inc = 1;
            } else if (A[i] > A[i + 1]) {
                if (inc == 1)
                    return false;
                inc = -1;
            }
        }
        return true;
    }
};

File: LeetCode/python3/896.py
class Solution:
    def isMonotonic(self, A):
        """
        :type A: List[int]
        :rtype: bool
        """
        inc = dec = False
        for i in range(len(A) - 1):
            if A[i] < A[i+1]:
                inc = True
            elif A[i] > A[i+1]:
                dec = True
        return not (inc and dec)
